vochain/ist/results.go

```diff
@@ -1,113 +1,24 @@
 package ist
 
 import (
-	"errors"
 	"fmt"
-	"time"
 
-	"go.vocdoni.io/dvote/db"
 	"go.vocdoni.io/dvote/log"
 	"go.vocdoni.io/dvote/vochain/results"
 )
 
 const (
-	// BlocksToWaitForResultsFactor is the factor used to compute the
+	// BlocksToWaitForResults is the factor used to compute the
 	// height at which the results will be committed to the state.
-	// The factor represents the number of votes required to compute
-	// in the time window of one blockchain block.
-	// For example, if the factor is 500, and there are 1000 votes,
-	// the results will be committed two blocks after.
 	// Note that changing this factor might break consensus and backwards
 	// compatibility.
-	BlocksToWaitForResultsFactor = 5000
-
-	// ExtraWaitSecondsForResults is the number of seconds extra to wait for the results
-	ExtraWaitSecondsForResults = time.Second * 20
+	BlocksToWaitForResults = 2
 )
 
-func (c *Controller) scheduleCommitResults(electionID []byte) error {
-	commitHeight, err := c.computeResultsCommitHeight(electionID)
-	if err != nil {
-		return fmt.Errorf("cannot compute results commit height: %w", err)
-	}
-	if err := c.Schedule(commitHeight, electionID, Action{
-		ID:         ActionCommitResults,
-		ElectionID: electionID,
-	}); err != nil {
-		return fmt.Errorf("cannot schedule results commit: %w", err)
-	}
-	return nil
-}
-
-// computeAndStoreResults computes the results for an election.
-// The results will be stored in the no-state once available.
-func (c *Controller) computeAndStoreResults(electionID []byte) error {
-	r, err := results.ComputeResults(electionID, c.state)
-	if err != nil {
-		return fmt.Errorf("cannot compute results: %w", err)
-	}
-	rEncoded, err := r.Encode()
-	if err != nil {
-		return fmt.Errorf("cannot encode results: %w", err)
-	}
-	if err := c.storeToNoState(dbResultsIndex(electionID), rEncoded); err != nil {
-		return fmt.Errorf("cannot store results: %w", err)
-	}
-	return nil
-}
-
 // commitResults commits the results to the state. If the `r` parameter is nil, it will
```

Comment:
Please update the comment for this function. It still tries to get the results from the no-state?

Reply:
If does not, probably the function must check that the `r` parameter is not `nil`